src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -615,6 +618,9 @@ ExecuteSingleSelectTask(CitusScanState *scanState, Task *task)
 		connection = GetPlacementListConnection(connectionFlags, placementAccessList,
 												NULL);
 
+		/* send BEGIN and assign distributed transaction ID */
+		RemoteTransactionBeginIfNecessary(connection);
```

Comment:
True, this only covers a very specific case where a coordinated transaction has been started, and all connections to the node are claimed exclusively such that a new connection was opened in `GetPlacementListConnection`, which currently can only happen with nested CTEs (because a coordinated transaction is started and the initial connection is claimed exclusively for copying the intermediate result). I'll add a comment.

I agree that starting a coordinated transaction for a plain router SELECT is not desirable.